### PR TITLE
Support DD env vars in infraattributesprocessor

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?
Adds support for Unified Service Tagging DD env vars in the infraattributesprocessor for the OTel Agent (`DD_ENV`, `DD_SERVICE`, `DD_VERSION`). This is done by adding the OTel semantic attribute in resource attributes when the tagger collects one of the following tags: `service, env, version`. This PR makes the following changes:
- Refactor common functions into a separate file in infraattributesprocessor
- Creates module for `comp/core/tagger/tags` in order to import constants
- Adds e2e test for this scenario in the infra attributes EKS test

### Motivation
OTEL-2140

### Describe how to test/QA your changes
Validation is done in e2e test, but can also be done manually by deploying the OTel Agent and an application with the DD env vars set and verifying that the unified service tags are correctly set in the UI.

### CODE REVIEW

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/common.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/collector/semconv/v1.21.0"
+
+	"github.com/DataDog/datadog-agent/comp/core/tagger/tags"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+)
+
+var unifiedServiceTagMap = map[string]string{
+	tags.Service: conventions.AttributeServiceName,
+	tags.Env:     conventions.AttributeDeploymentEnvironment,
```
Comment (liustanley) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/common.go:21]: @dineshg13 should we be preventing the override for `deployment.environment` or `deployment.environment.name` here? Or should we be doing both

Comment (dineshg13) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/common.go:21]: We should do mapping for both.

Comment (dineshg13) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/common.go:21]: I would create OTel -> DD semantics map as multiple Otel semantics can map to single DD semantic

**Review by purple4reina — APPROVED** [review]

File: test/new-e2e/tests/otel/utils/pipelines_utils.go [hunk 2]
```diff
@@ -617,6 +581,66 @@ func createCalendarApp(ctx context.Context, s OTelTestSuite) {
...
+		}}...)
+	}
+
+	return append(envVars, []corev1.EnvVar{{
+		Name:  "OTEL_SERVICE_NAME",
+		Value: calendarService,
+	}, {
+		Name: "OTEL_RESOURCE_ATTRIBUTES",
+		Value: resourceAttrs +
+			",service.name=$(OTEL_SERVICE_NAME)," +
```
Comment (dineshg13) [inline test/new-e2e/tests/otel/utils/pipelines_utils.go:638]: nit: u don't need to set in resource attrs once you have set `OTEL_SERVICE_NAME`

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/traces.go [hunk 3]
```diff
@@ -68,7 +68,18 @@ func (iasp *infraAttributesSpanProcessor) processTraces(_ context.Context, td pt
...
+			if !ust {
+				resourceAttributes.PutStr(k, v)
+				continue
+			}
+
+			// Add OTel semantics for unified service tags which are required in mapping
+			_, hasOTelAttr := resourceAttributes.Get(otelAttr)
+			if !hasOTelAttr {
+				resourceAttributes.PutStr(otelAttr, v)
+				continue
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/traces.go:81]: redundant

**Review by dineshg13 — COMMENTED** [review]
Few small nits

**Review by dustmop — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
👍🏻 LGTM for file owned by @DataDog/container-integrations.

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

File: tasks/modules.py [hunk 4]
```diff
@@ -148,6 +148,7 @@ def dependency_path(self, agent_version):
     "comp/core/secrets": GoModule("comp/core/secrets", independent=True, used_by_otel=True),
     "comp/core/status": GoModule("comp/core/status", independent=True, used_by_otel=True),
     "comp/core/status/statusimpl": GoModule("comp/core/status/statusimpl", independent=True),
+    "comp/core/tagger/tags": GoModule("comp/core/tagger/tags", independent=True, used_by_otel=True),
```
Comment (CelianR) [inline tasks/modules.py:151]: Note: This will be changed by https://github.com/DataDog/datadog-agent/pull/30753 soon (FYI if it gets merged first)

**Review by CelianR — APPROVED** [review]
LGTM for devx-infra files

**Review by KevinFairise2 — APPROVED** [review]

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics.go [hunk 5]
```diff
@@ -124,7 +68,23 @@ func (iamp *infraAttributesMetricProcessor) processMetrics(_ context.Context, md
...
+			hasOTelAttr := false
+			for _, otelAttr := range otelAttrs {
+				if _, ok := resourceAttributes.Get(otelAttr); ok {
+					hasOTelAttr = true
+					break
+				}
+			}
+			if !hasOTelAttr {
+				resourceAttributes.PutStr(otelAttrs[0], v)
+			}
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/metrics.go:87]: can we extract this in a common function and share across 3 signals

**Review by pducolin — APPROVED** [review]

**Review by jeremy-hanna — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**